### PR TITLE
[Notification service] BaseEvent 添加sender字段

### PR DESCRIPTION
**Pull request by wuchaochen**
## What is the purpose of the change

Notification service BaseEvent 对象添加sender字段代表发送事件的标识。

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests.

### CODE REVIEW

File: flink-ai-flow/lib/airflow/airflow/models/baseoperator.py [hunk 1]
```diff
@@ -137,14 +137,18 @@ def __init__(self, event_handler: EventHandler = None):
...
-    def subscribe_event(self, event_key: str, event_type: str = UNDEFINED_EVENT_TYPE, event_namespace: str = 'default'):
+    def subscribe_event(self, event_key: str,
+                        event_type: str = UNDEFINED_EVENT_TYPE,
+                        event_namespace: str = 'default',
+                        from_task_id: str = ''):
         """
         Subscribe to the events with the event_key and event_type. The event_handler will only handle the
         event that the operator subscribes to. event_type is optional, if it is not specify, it subscribes to all type
         of the event.
+        :param from_task_id: which task send the event
```
Comment (SteNicholas) [inline flink-ai-flow/lib/airflow/airflow/models/baseoperator.py:151]: The comment of `from_task_id` should be after the `event_namespace `.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: flink-ai-flow/lib/airflow/airflow/models/baseoperator.py [hunk 2]
```diff
@@ -153,9 +157,9 @@ def subscribe_event(self, event_key: str, event_type: str = UNDEFINED_EVENT_TYPE
...
         :return: None
         """
-        self._subscribed_events.add((event_namespace, event_key, event_type))
+        self._subscribed_events.add((event_namespace, event_key, event_type, from_task_id))
 
-    def get_subscribed_events(self) -> Set[Tuple[str, str, str]]:
+    def get_subscribed_events(self) -> Set[Tuple[str, str, str, str]]:
         """
         :return: the set of events that the operator subscribes to.
         :rtype: Set[str, str, str]
```
Comment (SteNicholas) [inline flink-ai-flow/lib/airflow/airflow/models/baseoperator.py:165]: The comment of the return type should update to `Set[str, str, str, str]`.

**Comment by SteNicholas** [comment]
@wuchaochen , notification service has already updated, therefore the Java client of notification service should sync the change of service.

File: flink-ai-flow/lib/notification_service/notification_service/client.py [hunk 3]
```diff
@@ -144,7 +148,9 @@ def send_event(self, event: BaseEvent):
                 value=event.value,
                 event_type=event.event_type,
                 context=event.context,
-                namespace=self._default_namespace if event.namespace is None else event.namespace),
+                namespace=self._default_namespace,
```
Comment (jiangxin369) [inline flink-ai-flow/lib/notification_service/notification_service/client.py:151]: 这里写死_default_namespace意思是client只能发自己namespace的event了吗？如果是的话_default_namespace在构造函数里默认值为None有点不合理，用default或者unassigned之类的是不是好一点，不然保存到数据库里之后会有歧义

Comment (SteNicholas) [reply flink-ai-flow/lib/notification_service/notification_service/client.py:151]: Notification client only send the events of the namespace from the constructor. And notification client could listen events from different namespaces.

**Review by jiangxin369 — APPROVED** [review]

**Review by SteNicholas — APPROVED** [review]

**Pull request merged**